server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1982,4 +1987,9 @@ public RefreshResult(boolean refreshed) {
             this(refreshed, UNKNOWN_GENERATION);
         }
     }
+
+    // Checks whether the given doc id (provided as uid) exists in the versionMap
+    public boolean isInVersionMap(BytesRef uid) {
```

Comment:
As I mentioned in my other comment, this is okay if necessary. But I don't love adding more methods to the engine if this can be avoided.